maplibre_gl_example/lib/layer_manipulation.dart

```diff
@@ -0,0 +1,327 @@
+    );
+  }
+
+  Future<void> _setLayerFilter() async {
+    if (mapController == null) return;
+
+    // Cycle through different filters
+    final filters = [
+      '["all", ["==", "name", "International Date Line"]]', // Don't show International Date Line
+      '["all", ["!=", "name", "International Date Line"]]', // Do show International Date Line
+    ];
+
+    final filter = filters[currentFilterId % filters.length];
+    await mapController!.setLayerFilter("countries-fill", filter);
+
+    setState(() {
+      currentFilterId++;
+    });
+  }
+
+  @override
+  void dispose() {
+    _animationTimer?.cancel();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: Column(
+        children: [
+          // Map
+          Expanded(
+            flex: 3,
+            child: MapLibreMap(
+              styleString: MapLibreStyles.demo,
+              onMapCreated: _onMapCreated,
+              onStyleLoadedCallback: _onStyleLoadedCallback,
+              initialCameraPosition: const CameraPosition(
+                target: LatLng(50.0, 10.0),
+                zoom: 3.0,
+              ),
+            ),
+          ),
+
+          // Controls
+          Expanded(
+            flex: 2,
+            child: SingleChildScrollView(
+              padding: const EdgeInsets.all(16.0),
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Text(
+                    'Layer Manipulation Demo',
+                    style: Theme.of(context).textTheme.headlineSmall,
+                  ),
+                  const SizedBox(height: 16),
+
+                  // Control buttons
+                  Wrap(
+                    spacing: 8.0,
+                    runSpacing: 8.0,
+                    children: [
+                      ElevatedButton(
+                        onPressed: _toggleAnimation,
+                        style: ElevatedButton.styleFrom(
+                          backgroundColor:
+                              _isAnimating ? Colors.red : Colors.green,
+                          foregroundColor: Colors.white,
+                        ),
+                        child: Text(_isAnimating
+                            ? 'Stop Animation'
+                            : 'Start Animation'),
+                      ),
+                      ElevatedButton(
+                        onPressed: _editGeoJsonUrl,
+                        child: const Text('Show Earthquakes'),
+                      ),
+                      ElevatedButton(
+                        onPressed: _setLayerFilter,
+                        child: const Text('Toggle Country Fill'),
+                      ),
+                      ElevatedButton(
+                        onPressed: _getCurrentStyle,
+                        child: const Text('Get Style'),
+                      ),
```

Comment:
These methods aren't available on web, disable them when `kIsWeb`
```suggestion
                    if(!kIsWeb) ...[
                        ElevatedButton(
                            onPressed: _editGeoJsonUrl,
                            child: const Text('Show Earthquakes'),
                        ),
                        ElevatedButton(
                            onPressed: _setLayerFilter,
                            child: const Text('Toggle Country Fill'),
                        ),
                        ElevatedButton(
                            onPressed: _getCurrentStyle,
                            child: const Text('Get Style'),
                        ),
                    ],
```